### DISCUSSION TITLE
[DOCS] cdk-graph-plugin-diagram documentation could have quickstart documentation

### DISCUSSION THREAD

**Discussion by miiiak** [aws/aws-pdk#291] [Q&A]
### Describe the issue

Hey there, 

I'm new to typescript.  I followed the guidance in https://github.com/aws/aws-prototyping-sdk/blob/mainline/README.md, then moved to https://github.com/aws/aws-prototyping-sdk/tree/mainline/packages/cdk-graph-plugin-diagram, as it was my primary purpose to experiment making CDK based graphs.

Issues I encountered:
- I don't actually know how to run my typescript file.  I chose visual studio code.
- What is IIFE?
- Can you explain to me how to point to a CDK folder/file, and load it? Quickstart line ` // ... add stacks, etc` is not helpful to the noob. 
- There's a lot of implicit knowledge here that I lack.  What do I need to do to get this to run if I am starting from scratch? Reference other guides where necessary.

### Links

https://github.com/aws/aws-prototyping-sdk/blob/mainline/README.md

**Comment by JeremyJonas** [comment]
Hi @miiiak, thank you for your interest in using this project and the experimental functionality. The project is still in "pre-release (v0)" and the cdk-graph/cdk-graph-plugins-diagram are definitely on the bleeding edge of experimental within this project. While we continue to evolve this project and the packages it provides toward v1 release, we will definitely take this recommendation into consideration.

There is a lot of documentation already available for CDK, Typescript, etc, and we don't expect to reproduce that documentation here, however I do agree we should point users such as your self in the right direction to ramp up on the necessary tooling to utilize this project.

From your perspective, what would your expectations be on this project to support filling those gaps while not reproducing?

For reference, here are some great resources:
- CDK Documentation: https://docs.aws.amazon.com/cdk/v2/guide/home.html
- CDK Workshops: https://cdkworkshop.com/
  - https://cdkworkshop.com/20-typescript/20-create-project/300-structure.html
- CDK Training Materials and Sample Code: https://github.com/kalaiser/awesome-cdk#training-materials-and-sample-code
- Typescript: https://www.typescriptlang.org/
- Javascript: https://www.javascript.com/learn
   - IIFE (Immediately Invoked Function Expression): https://developer.mozilla.org/en-US/docs/Glossary/IIFE

I hope this helps point you in the right direction to get going.
